### PR TITLE
Documentation update of MappingFEField.

### PR DESCRIPTION
**Pull request by bangerth**
In particular, group members into doxygen group. Like
#1382 but for MappingFEField.

In reference to #1198.

### CODE REVIEW

File: include/deal.II/fe/mapping_fe_field.h [hunk 1]
```diff
@@ -491,6 +445,51 @@ class MappingFEField : public Mapping<dim,spacedim>
...
+   * prevent construction in 1D and 2D, but since memory and time requirements
+   * are not particularly high this seems unnecessary at the moment.
+   */
+  SmartPointer<const FiniteElement<dim,spacedim>, MappingFEField<dim,spacedim,DH,VECTOR> > fe;
+
+
+  /**
+   * Pointer to the DoFHandler to which the mapping vector is associated.
+   */
+  SmartPointer<const DH,MappingFEField<dim,spacedim,DH,VECTOR> >euler_dof_handler;
```
Comment (tjhei) [inline include/deal.II/fe/mapping_fe_field.h]: missing space after "> >"?

File: include/deal.II/fe/mapping_fe_field.h [hunk 2]
```diff
@@ -491,6 +445,51 @@ class MappingFEField : public Mapping<dim,spacedim>
...
    */
 
+  /**
+   * This function and the next allow to generate the transform require by the
+   * virtual transform() in mapping, but unfortunately in C++ one cannot
+   * declare a virtual template function.
+   */
+  template <int rank>
+  void
+  transform_fields(const VectorSlice<const std::vector<Tensor<rank,dim>      > > input,
```
Comment (tjhei) [inline include/deal.II/fe/mapping_fe_field.h]: weird spacing

File: include/deal.II/fe/mapping_fe_field.h [hunk 3]
```diff
@@ -491,6 +445,51 @@ class MappingFEField : public Mapping<dim,spacedim>
...
+                   VectorSlice<      std::vector<Tensor<rank,spacedim> > > output,
+                   const typename Mapping<dim,spacedim>::InternalDataBase &internal,
+                   const MappingType type) const;
+
+
+  /**
+   * see doc in transform_fields
```
Comment (tjhei) [inline include/deal.II/fe/mapping_fe_field.h]: can you change this to "see transform_fields() above." or something?

File: include/deal.II/fe/mapping_fe_field.h [hunk 4]
```diff
@@ -115,54 +115,65 @@ class MappingFEField : public Mapping<dim,spacedim>
...
-   *
-   * Uses Newton iteration and the @p transform_unit_to_real_cell function.
-   *
-   * In the codimension one case, this function returns the normal projection
-   * of the real point @p p on the curve or surface identified by the @p cell.
+   * @name Mapping points between reference and real cells
+   * @{
+   */
+
+  // for documentation, see the Mapping base class
```
Comment (tjhei) [inline include/deal.II/fe/mapping_fe_field.h]: is there a reason you are not using doxygen comments here and below?

File: source/fe/mapping_fe_field.cc [hunk 5]
```diff
@@ -58,13 +59,83 @@ MappingFEField<dim,spacedim,VECTOR,DH>::InternalData::InternalData (const Finite
...
+MappingFEField<dim,spacedim,VECTOR,DH>::InternalData::
+memory_consumption () const
 {
   Assert (false, ExcNotImplemented());
   return 0;
 }
 
 
+
+template<int dim, int spacedim, class DH, class VECTOR>
```
Comment (tjhei) [inline source/fe/mapping_fe_field.cc]: You moved a bunch of functions to the .cc. Are any of these performance critical so inlining could matter?

File: include/deal.II/fe/mapping_fe_field.h [hunk 6]
```diff
@@ -562,20 +560,12 @@ class MappingFEField : public Mapping<dim,spacedim>
    */
   std::vector<unsigned int> fe_to_real;
 
-  /**
```
Comment (tjhei) [inline include/deal.II/fe/mapping_fe_field.h]: Do you consider this comment redundant?

Comment (bangerth) [inline include/deal.II/fe/mapping_fe_field.h]: No, it's just wrong and I don't have anything better right now. The function is simply not reimplemented from Mapping (Mapping has no such function) and I thought no comment is better than a wrong comment.

File: source/fe/mapping_fe_field.cc (hunk 5 above)
Comment (bangerth) [inline source/fe/mapping_fe_field.cc]: No. Mapping objects are only ever used through pointers to a base class, never (outside the implementation of the class itself) through the class itself. Consequently, there are no opportunities for inlining and so I thought it is safe to move everything.

In particular, these functions are part of the private interface of the class, so putting them into the .cc file keeps them available for inlining for all possible users (which all reside in the same file anyway).

File: include/deal.II/fe/mapping_fe_field.h (hunk 4 above)
Comment (bangerth) [inline include/deal.II/fe/mapping_fe_field.h]: If doxygen sees no doxygen-style comment in front of a virtual function, it copies the comment from the base class. This is exactly what I want -- using a doxygen-style comment here would defeat the purpose.

Of course, I could have copied the comment from the base class myself, but then I would have to keep the original and multiple copies in synch. Or I could have had no comment (of any style) here at all, but the perfectionist in me does not allow for that :-)

**Comment by bangerth** [comment]
I think I've addressed all of your comments.

**Pull request merged**